ci/snapcraft.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/sh
+
+set -ef
+
+# This script encapsulates the job of preparing snapcract configuration for the
+# rustup snap.  We construct either an edge snap (builds on PRs or master)
+# or a beta snap if built from a non-master branch.
+
+# If built on master, we should publish the edge snap
+# If built on stable, we should publish the beta snap
+
+# To parameterise the build, the following environment variables are needed:
+# DO_SNAP=1  <-- without this, we don't run
+# SNAP_EDGE=1 <-- with this, we build an edge/devel snap, otherwise a release snap
+# SNAP_ARCH=... <-- without this we cannot build, this is the snap architecture we're building
+
+if [ -z "$DO_SNAP" ]; then
```

Comment:
fixing this for undefined safety is easy:
```
if [ -z "${DO_SNAP:-}" ]; then
```

Reply:
Of course.  I'm so dozy when it comes to shell :D

Reply:
Done this where appropriate, thanks Robert.